balancer-labs/frontend-v2
# Pool creation unknown tokens

# Description

Currently the pool creation UI does not support arbitrage warnings and optimisations for tokens which may not have a current price on coingecko. This update adds the ability for the user to add in those prices themselves.

## Type of change


## How should this be tested?

Here is a custom token: `0xbF3122A7a98a2F9971C339a8511378d853585876`
Basically adding a custom token which doesn't have a price on coingecko to the pool creation token list should trigger a modal to pop up requesting the user to add the price for that token.
Once you enter that price the remainder of the pool creation flow should work as expected.

A quick way to check that the price has been injected correctly is to type paste the token address back into the token select modal and you should see a price there, compared to a '-'

## Checklist:
